src/main/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -597,77 +620,16 @@ private FingerprintIdDetails createUnknownFP(final Path samFile, final SAMRecord
      * Generates a per-sample Fingerprint for the contaminant in the supplied SAM file.
      * Data is aggregated by sample, not read-group.
      */
-    public Map<String, Fingerprint> identifyContaminant(final Path samFile, final double contamination, final int locusMaxReads) {
-        final Map<String, Fingerprint> fingerprintsBySample = new HashMap<>();
-
-        try (final SamReader in = SamReaderFactory.makeDefault().enable(CACHE_FILE_BASED_INDEXES).open(samFile)) {
-            checkDictionaryGoodForFingerprinting(in.getFileHeader().getSequenceDictionary());
-
-            final SamLocusIterator iterator = new SamLocusIterator(in, haplotypes.getIntervalList(), in.hasIndex());
-            iterator.setEmitUncoveredLoci(true);
-            iterator.setMappingQualityScoreCutoff(this.minimumMappingQuality);
-            iterator.setQualityScoreCutoff(this.minimumBaseQuality);
-
-            // In some cases it is useful to allow duplicate reads to be used - the most common is in single-end
-            // sequence data where the duplicate marking may have been overly aggressive, and there is useful
-            // non-redundant data in the reads marked as "duplicates'.
-            if (this.allowDuplicateReads) {
-                final List<SamRecordFilter> filters = new ArrayList<>(1);
-                filters.add(new SecondaryAlignmentFilter());
-                iterator.setSamFilters(filters);
-            }
+    public Map<String, Fingerprint> identifyContaminant(final Path samFile, final double contamination) {
 
-            for (final SAMReadGroupRecord rg : in.getFileHeader().getReadGroups()) {
-                if (!fingerprintsBySample.containsKey(rg.getSample())) {
-                    final Fingerprint fingerprint = new Fingerprint(rg.getSample(),
-                            samFile,
-                            rg.getSample());
+        final Map<FingerprintIdDetails, Fingerprint> fingerprintIdDetailsFingerprintMap = this.fingerprintSamFile(samFile, (h) -> new HaplotypeProbabilitiesFromContaminatorSequence(h, contamination));
```

Comment:
`()`s unnecessary for single arg lambda
```suggestion
        final Map<FingerprintIdDetails, Fingerprint> fingerprintIdDetailsFingerprintMap = this.fingerprintSamFile(samFile, h -> new HaplotypeProbabilitiesFromContaminatorSequence(h, contamination));
```